src/Solve.jl

```diff
@@ -535,6 +624,83 @@ function _can_solve_internal_no_check(C::SolveCtx{T}, b::MatElem{T}, task::Symbo
   return true, sol, kernel(C, side = side)
 end
 
+################################################################################
+#
+#  Special internal functonality
+#
+################################################################################
+
+# _can_solve_internal_no_check methods which only work or are efficient for
+# certain types
+
+function _common_denominator(A::MatElem{T}) where T <: Union{FracElem, Rational{BigInt}}
+  d = numerator(one(base_ring(A)))
+  @inbounds for j in 1:ncols(A)
+    for i in 1:nrows(A)
+      d = lcm(d, denominator(A[i, j]))
+    end
+  end
+  return d
+end
+
+# The fflu approach is the fastest over a fraction field (see benchmarks on PR 661)
+function _can_solve_internal_no_check(A::MatElem{T}, b::MatElem{T}, task::Symbol; side::Symbol = :left) where T <: Union{FracElem, Rational{BigInt}}
```

Comment:
Is this also used for the `SolveContext` of these base rings?

Reply:
Now it is. (I will need to add more tests when the code coverage is in, let's see.)